util/src/main/java/io/grpc/util/LruCache.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2024 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.util;
```

Comment:
Util isn't a random utility dumping ground. The regular utility dumping ground is `io.grpc.internal`. `io.grpc.util` doesn't have a strong identity, but it is some utilities for our _users_ that didn't make sense to go in `io.grpc` directly.